packages/blade/src/components/Box/__tests__/styled-props.web.test.tsx

```diff
@@ -0,0 +1,70 @@
+import styled from 'styled-components';
+import BaseBox from '../BaseBox';
+import type { StyledProps } from '~components/Box/styled-props';
+import { useStyledProps, getStyledProps } from '~components/Box/styled-props';
+import renderWithTheme from '~src/_helpers/testing/renderWithTheme.web';
+
+type MyComponentWithStyledProps = {
+  test: 'working???';
+} & StyledProps;
+
+const ComponentWithGetStyledProps = (props: MyComponentWithStyledProps): JSX.Element => {
+  return <BaseBox {...getStyledProps(props)}>Hello!</BaseBox>;
+};
+
+const ComponentWithUseStyledProps = styled.button<MyComponentWithStyledProps>((props) => {
+  const styledPropsCSSObject = useStyledProps(props);
```

Comment:
i think we shall add similar test for `useMemoisedStyles`?

Reply:
This itself tests useMemoizedStyles since it is used by `useStyledProps`.

I am not really sure how I can unit-test that hook since it has to be used inside another mock hook or mock component which will eventually end us with same test as this one.

Reply:
added one small mock test in upcoming commit.